### PR TITLE
Feat/public view sharing

### PR DESCRIPTION
**Pull request by Crash--**
I splited the public toolbar in 2 components: 
- one for a sharing by link 
- one for a cozy to cozy sharing.

This was done because we had "desktopbtoolbar / cozytoolbar / mobiletoolbar" with a lot of conditions inside and it was difficult to understand and follow. 

Now we only have two components, pretty close to the functional specifications. It's easier to understand and edit. 

I hope that each commit is self explanatory

### CODE REVIEW

File: src/drive/locales/en.json [hunk 1]
```diff
@@ -87,6 +90,14 @@
         "action": "Authorize access",
         "success": "The application has access to your contacts"
       }
+    },
+    "banner": {
+      "shared_from": "Shared from the cozy of",
```
Comment (ptbrowne) [inline src/drive/locales/en.json:95]: I think it'd be better English to have `Shared from %{name}'s cozy`.

File: src/drive/locales/en.json [hunk 2]
```diff
@@ -87,6 +90,14 @@
...
+      "synchronise": "Synchronise this content in your cozy to be able to modify it or to keep it in case of cancellation of sharing.",
+      "whats_cozy": "Cozy is a secure digital home for all your data, hosted in France - ",
+      "know_more": "More information",
+      "sync_to_mine": "Synchronise in my Cozy",
```
Comment (ptbrowne) [inline src/drive/locales/en.json:99]: I think "Sync to my Cozy" would be better.

**Review by ptbrowne — APPROVED** [review]

File: src/drive/web/modules/public/OpenInCozyButton.jsx [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
 
-const OpenInCozyButton = ({ t, size = 'normal', href = '' }) => (
-  <ButtonLink
-    href={href}
-    size={size}
-    label={t('toolbar.menu_open_cozy')}
-    icon={'to-the-cloud'}
-  />
-)
+const OpenInCozyButton = ({ size = 'normal', href = '' }) => {
```
[the remark this answers is not in the document]
Comment (Crash--) [reply src/drive/web/modules/public/OpenInCozyButton.jsx:4]: Oh something missed in my squash / edit / rebase / bisect work... 

The change are made in https://github.com/cozy/cozy-drive/commit/ebb95b406063bdcda55dbcc87690a577e362ed72

File: src/drive/web/modules/public/DownloadButton.jsx [hunk 4]
```diff
@@ -1,8 +1,31 @@
...
+    <DownloadButton
+      label={t('toolbar.menu_download')}
+      data-test-id="fil-public-download"
+      onDownload={() => {
+        downloadFiles(client, files)
+      }}
+      theme="secondary"
+    />
+  )
+}
```
[the remark this answers is not in the document]
Comment (Crash--) [reply src/drive/web/modules/public/DownloadButton.jsx:28]: Yep later in https://github.com/cozy/cozy-drive/commit/d308c4bbefdca1934bab7a1057c8d93293eeeed4#diff-fb904c2f707f699be7e8837214018fcae1fc04eacfce1d714ca33d1e3204ef27R16 

Next time, I'll squash the creation and the use together

File: src/drive/locales/en.json (hunk 1 above)
Comment (Crash--) [reply src/drive/locales/en.json]: 783eaa6ef7705148c2eeb0b44e08aba34aeef3de

File: src/drive/locales/en.json (hunk 2 above)
Comment (Crash--) [reply src/drive/locales/en.json:99]: 783eaa6ef7705148c2eeb0b44e08aba34aeef3de

File: src/drive/web/modules/public/PublicToolbar.jsx [hunk 5]
```diff
@@ -232,9 +232,25 @@ const DesktopToolbar = (
...
 
+       /**
+ * Returns if the shortcut for the sharing is created on the recipient's cozy
+ * 
+ * To check that, if the `instance` attribute is set to the sharing member then
+ * we know that the shortcut is already created 
+ * 
+ * @param {Array} included 
+ */
+const hasSharingShorcutCreated = (included) => {
```
[the remark this answers is not in the document]
Comment (Crash--) [reply src/drive/web/modules/public/PublicToolbar.jsx]: b735ee2

File: src/drive/web/modules/public/PublicToolbar.jsx [hunk 6]
```diff
@@ -232,9 +232,25 @@ const DesktopToolbar = (
 
 DesktopToolbar.propTypes = toolbarProptypes
 
+       /**
+ * Returns if the shortcut for the sharing is created on the recipient's cozy
+ * 
+ * To check that, if the `instance` attribute is set to the sharing member then
+ * we know that the shortcut is already created
```
[the remark this answers is not in the document]
Comment (Crash--) [reply src/drive/web/modules/public/PublicToolbar.jsx]: b735ee2

**Pull request merged**